Documentation Updates (#155)
techservicesillinois/SecOps-Powershell-CISDSC: docs/dsc_getting_started.md

# What setup is needed to run a configuration?

> - As is this document is written for using DSC standalone. If using these resources in another product such as Ansible or Puppet consult the vendor's documentation.

Reviewer: ```suggestion
- This document is written for using DSC alone. If using these resources in another product such as Ansible or Puppet consult the vendor's documentation.
```

Author: agreed and fixed

Author: oh I didnt see I could commit straight from your suggestion neat. I pushed the update from my laptop.

### Enabling

> - WinRM will need to be enabled on the machine for the LCM to do it's job. This is due to it leveraging "implicit remoting". Documentation for WinRM can be found [here](https://docs.microsoft.com/en-us/windows/win32/winrm/installation-and-configuration-for-windows-remote-management) or you can use the quick setups with either of the following commands to enable with default configuration. These commands will not work if any network adapter is set to use the public network profile. It is possible to bypass this but not recommended for security reasons, recommended solution is to set them to domain or private. This includes virtual adapters such as ones created by VPNs or Hypver-V.

Reviewer: ```suggestion
- WinRM will need to be enabled on the machine for the LCM to do its job. This is due to it leveraging "implicit remoting". Documentation for WinRM can be found [here](https://docs.microsoft.com/en-us/windows/win32/winrm/installation-and-configuration-for-windows-remote-management), or you can use the quick setups with either of the following commands to enable it with a default configuration. These commands will not work if any network adapter is set to use the _Public_ network profile. If any network adapter does not have the "Public" network profile, the recommended solution is to set it to _Domain_ or _Private_. This includes virtual adapters such as ones created by VPNs or Hypver-V.
```

Reviewer: I was able to confirm about WinRM not working with a Public network profile, but not that virtual adapters were included. Are you sure about that one?

Author: Yeah I ran into this during a demo with EPS actually lol. Profiles are adapter specific and my hyper-v adapter was stuck on public.

Reviewer: That might be for Hyper-V only. Aside from the VPN adapter, my VMware virtual adapters don't have a network profile at all.

Reviewer: Regardless, I'd add the changes I suggested for clarity.

```powershell
Set-Item -Path 'WSMan:\localhost\MaxEnvelopeSizeKb' -Value 8192
```

## LCM
- The LCM is what executes DSC tasks on Windows and has been pre-installed since Windows 7. LCM configuration will be subject to your specific environment and needs. Below is an example of configuring it to act in an ad-hoc execution only mode disabling any routine consistency checks or [pull server](https://docs.microsoft.com/en-us/powershell/scripting/dsc/pull-server/pullserver?view=powershell-7) check ins. Advanced configuration options can be found [here](https://docs.microsoft.com/en-us/powershell/scripting/dsc/managing-nodes/metaconfig?view=powershell-7).

```powershell

> [DSCLocalConfigurationManager()]

Reviewer: If the idea is that this is the LCM configuration file that you first create, then apply, then remove, you should split this into two or three separate code blocks. It's rather confusing to have the file text and the code that uses the file together.

Author: Maybe the explanation just need more clarification? I'd always expect the code to be ran like the example in one chunk.

Author: addressed in last push

Reviewer: You should probably have a line before each code block to explain what each is. (ex: "Create a file with this as the contents.") Also, is the final line in the first block supposed to be part of that file's contents?
